### PR TITLE
automate arlon installation

### LINKED ISSUE #295: Install arlon(the controller) using arlon(the CLI)

### PR DESCRIPTION
**Pull request by Rohitrajak1807**
Fixes #295 

Running with `-y` flag to accept the defaults(useful for scripts)
```shell
➜  arlon git:(private/Rohitrajak1807/223) bin/arlon init -y
Cannot find argocd-server deployment. Argocd may not be installed
successfully created (apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) argocd/applications.argoproj.io
successfully created (apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) argocd/applicationsets.argoproj.io
successfully created (apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) argocd/appprojects.argoproj.io
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-application-controller
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-applicationset-controller
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-dex-server
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-notifications-controller
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-redis
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-repo-server
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-server
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-application-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-applicationset-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-dex-server
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-notifications-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-server
successfully created (rbac.authorization.k8s.io/v1, Kind=ClusterRole) argocd/argocd-application-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=ClusterRole) argocd/argocd-server
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-application-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-applicationset-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-dex-server
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-notifications-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-redis
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-server
successfully created (rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) argocd/argocd-application-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) argocd/argocd-server
successfully created (/v1, Kind=ConfigMap) argocd/argocd-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-cmd-params-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-gpg-keys-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-notifications-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-rbac-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-ssh-known-hosts-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-tls-certs-cm
successfully created (/v1, Kind=Secret) argocd/argocd-notifications-secret
successfully created (/v1, Kind=Secret) argocd/argocd-secret
successfully created (/v1, Kind=Service) argocd/argocd-applicationset-controller
successfully created (/v1, Kind=Service) argocd/argocd-dex-server
successfully created (/v1, Kind=Service) argocd/argocd-metrics
successfully created (/v1, Kind=Service) argocd/argocd-notifications-controller-metrics
successfully created (/v1, Kind=Service) argocd/argocd-redis
successfully created (/v1, Kind=Service) argocd/argocd-repo-server
successfully created (/v1, Kind=Service) argocd/argocd-server
successfully created (/v1, Kind=Service) argocd/argocd-server-metrics
successfully created (apps/v1, Kind=Deployment) argocd/argocd-applicationset-controller
successfully created (apps/v1, Kind=Deployment) argocd/argocd-dex-server
successfully created (apps/v1, Kind=Deployment) argocd/argocd-notifications-controller
successfully created (apps/v1, Kind=Deployment) argocd/argocd-redis
successfully created (apps/v1, Kind=Deployment) argocd/argocd-repo-server
successfully created (apps/v1, Kind=Deployment) argocd/argocd-server
successfully created (apps/v1, Kind=StatefulSet) argocd/argocd-application-controller
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-application-controller-network-policy
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-dex-server-network-policy
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-redis-network-policy
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-repo-server-network-policy
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-server-network-policy
waiting for argocd-server
waiting for argocd-server
waiting for argocd-server
waiting for argocd-server
waiting for argocd-server
waiting for argocd-server
waiting for argocd-server
waiting for argocd-server
Forwarding from 127.0.0.1:8080 -> 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
'admin:login' logged in successfully
Context '127.0.0.1:8080' updated
Handling connection for 8080
namespage arlon created
ConfigMap argocd-cm updated
ConfigMap argocd-rbac-cm updated
Handling connection for 8080
Secret argocd-creds createdsuccessfully created (apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) arlon/profiles.core.arlon.io
successfully created (apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) arlon/clusterregistrations.core.arlon.io
successfully created (apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) arlon/callhomeconfigs.core.arlon.io
successfully created (apps/v1, Kind=Deployment) arlon/arlon-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=ClusterRole) arlon/callhomeconfig-updater
successfully created (rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) arlon/arlon-callhomeconfig-writer
successfully created (rbac.authorization.k8s.io/v1, Kind=ClusterRole) arlon/clusterregistration-updater
successfully created (rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) arlon/arlon-clusterregistration-writer
successfully created (apps/v1, Kind=Deployment) arlon/arlon-webhook
waiting for arlon-controller
waiting for arlon-controller
waiting for arlon-controller
waiting for arlon-controller
➜  arlon git:(private/Rohitrajak1807/223) 
```

Running in interactive mode(default)

```shell
➜  arlon git:(private/Rohitrajak1807/223) kubectl delete ns arlon && kubectl delete ns argocd && bin/arlon init    
namespace "arlon" deleted
namespace "argocd" deleted
Cannot find argocd-server deployment. Argocd may not be installed
argo-cd not found, possibly not installed. Proceed to install? [y/n]y
(apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) argocd/applications.argoproj.io already exists
(apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) argocd/applicationsets.argoproj.io already exists
(apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) argocd/appprojects.argoproj.io already exists
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-application-controller
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-applicationset-controller
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-dex-server
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-notifications-controller
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-redis
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-repo-server
successfully created (/v1, Kind=ServiceAccount) argocd/argocd-server
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-application-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-applicationset-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-dex-server
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-notifications-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=Role) argocd/argocd-server
(rbac.authorization.k8s.io/v1, Kind=ClusterRole) argocd/argocd-application-controller already exists
(rbac.authorization.k8s.io/v1, Kind=ClusterRole) argocd/argocd-server already exists
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-application-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-applicationset-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-dex-server
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-notifications-controller
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-redis
successfully created (rbac.authorization.k8s.io/v1, Kind=RoleBinding) argocd/argocd-server
(rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) argocd/argocd-application-controller already exists
(rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) argocd/argocd-server already exists
successfully created (/v1, Kind=ConfigMap) argocd/argocd-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-cmd-params-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-gpg-keys-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-notifications-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-rbac-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-ssh-known-hosts-cm
successfully created (/v1, Kind=ConfigMap) argocd/argocd-tls-certs-cm
successfully created (/v1, Kind=Secret) argocd/argocd-notifications-secret
successfully created (/v1, Kind=Secret) argocd/argocd-secret
successfully created (/v1, Kind=Service) argocd/argocd-applicationset-controller
successfully created (/v1, Kind=Service) argocd/argocd-dex-server
successfully created (/v1, Kind=Service) argocd/argocd-metrics
successfully created (/v1, Kind=Service) argocd/argocd-notifications-controller-metrics
successfully created (/v1, Kind=Service) argocd/argocd-redis
successfully created (/v1, Kind=Service) argocd/argocd-repo-server
successfully created (/v1, Kind=Service) argocd/argocd-server
successfully created (/v1, Kind=Service) argocd/argocd-server-metrics
successfully created (apps/v1, Kind=Deployment) argocd/argocd-applicationset-controller
successfully created (apps/v1, Kind=Deployment) argocd/argocd-dex-server
successfully created (apps/v1, Kind=Deployment) argocd/argocd-notifications-controller
successfully created (apps/v1, Kind=Deployment) argocd/argocd-redis
successfully created (apps/v1, Kind=Deployment) argocd/argocd-repo-server
successfully created (apps/v1, Kind=Deployment) argocd/argocd-server
successfully created (apps/v1, Kind=StatefulSet) argocd/argocd-application-controller
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-application-controller-network-policy
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-dex-server-network-policy
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-redis-network-policy
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-repo-server-network-policy
successfully created (networking.k8s.io/v1, Kind=NetworkPolicy) argocd/argocd-server-network-policy
waiting for argocd-server
waiting for argocd-server
waiting for argocd-server
waiting for argocd-server
waiting for argocd-server
arlon namespace not found. Install arlon controller?[y/n]y
Forwarding from 127.0.0.1:8080 -> 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
'admin:login' logged in successfully
Context '127.0.0.1:8080' updated
Handling connection for 8080
namespage arlon created
ConfigMap argocd-cm updated
ConfigMap argocd-rbac-cm updated
Handling connection for 8080
Secret argocd-creds created(apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) arlon/profiles.core.arlon.io already exists
(apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) arlon/clusterregistrations.core.arlon.io already exists
(apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) arlon/callhomeconfigs.core.arlon.io already exists
successfully created (apps/v1, Kind=Deployment) arlon/arlon-controller
(rbac.authorization.k8s.io/v1, Kind=ClusterRole) arlon/callhomeconfig-updater already exists
(rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) arlon/arlon-callhomeconfig-writer already exists
(rbac.authorization.k8s.io/v1, Kind=ClusterRole) arlon/clusterregistration-updater already exists
(rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) arlon/arlon-clusterregistration-writer already exists
successfully created (apps/v1, Kind=Deployment) arlon/arlon-webhook
waiting for arlon-controller
waiting for arlon-controller
waiting for arlon-controller
```

When `arlon` NS is absent, but argocd server deployment is present
```shell
➜  arlon git:(private/Rohitrajak1807/223) bin/arlon init   
arlon namespace not found. Install arlon controller?[y/n]y
Forwarding from 127.0.0.1:8080 -> 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
'admin:login' logged in successfully
Context '127.0.0.1:8080' updated
Handling connection for 8080
namespage arlon created
ConfigMap argocd-cm updated
ConfigMap argocd-rbac-cm updated
Handling connection for 8080
Secret argocd-creds created(apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) arlon/profiles.core.arlon.io already exists
(apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) arlon/clusterregistrations.core.arlon.io already exists
(apiextensions.k8s.io/v1, Kind=CustomResourceDefinition) arlon/callhomeconfigs.core.arlon.io already exists
successfully created (apps/v1, Kind=Deployment) arlon/arlon-controller
(rbac.authorization.k8s.io/v1, Kind=ClusterRole) arlon/callhomeconfig-updater already exists
(rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) arlon/arlon-callhomeconfig-writer already exists
(rbac.authorization.k8s.io/v1, Kind=ClusterRole) arlon/clusterregistration-updater already exists
(rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding) arlon/arlon-clusterregistration-writer already exists
successfully created (apps/v1, Kind=Deployment) arlon/arlon-webhook
waiting for arlon-controller
waiting for arlon-controller
waiting for arlon-controller
```
When arlon NS and argocd-server deployment are present

```shell
➜  arlon git:(private/Rohitrajak1807/223) bin/arlon init
➜  arlon git:(private/Rohitrajak1807/223) 
```

### CODE REVIEW

File: cmd/install/install.go [hunk 1]
```diff
@@ -238,7 +238,7 @@ func installArgoCDPlatform() error {
 // Downloads the latest version of argocd
 func downloadArgoCD(osPlatform string) error {
 	var downloadArgoCD string
-	argocdVersion := "v2.4.11"
+	argocdVersion := "v2.4.16"
```
Comment (cruizen) [inline cmd/install/install.go:241]:
```suggestion
	argocdVersion := "v2.4.17"
```

**Comment by Rohitrajak1807** [comment]
> Can you rebase on top of main once again? Some of these changes have already made it to main branch as part of other PRs.

Did this just now.... (after reading the comment)
```
➜  arlon git:(main) ✗ git pull            
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 3 (delta 0), reused 3 (delta 0), pack-reused 0
Unpacking objects: 100% (3/3), 946 bytes | 189.00 KiB/s, done.
From github.com:arlonproj/arlon
 + 7151634...34d9bd5 gh-pages   -> origin/gh-pages  (forced update)
Already up to date.
➜  arlon git:(main) ✗ git checkout private/Rohitrajak1807/223 
Switched to branch 'private/Rohitrajak1807/223'
Your branch is up to date with 'origin/private/Rohitrajak1807/223'.
➜  arlon git:(private/Rohitrajak1807/223) git rebase main                         
Current branch private/Rohitrajak1807/223 is up to date.
➜  arlon git:(private/Rohitrajak1807/223) 
```

**Review by cruizen — APPROVED** [review]
Looks good, thanks.

File: config/embed.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package config
+
+import _ "embed"
+
+var (
+	//go:embed crd/bases/core.arlon.io_callhomeconfigs.yaml
```
Comment (cruizen) [inline config/embed.go:6]: We have to remember to add new CRDs here e.g. upcoming AppProfile

File: .github/workflows/trivy.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@
 # separate terms of service, privacy policy, and support
 # documentation.
 
-name: build
+name: Trivy Code Scan
```
Comment (cruizen) [inline .github/workflows/trivy.yml:6]: Whoops, didn't notice this earlier. :-D

File: cmd/initialize/init.go [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+			kubeClient := kubernetes.NewForConfigOrDie(cfg)
+			canInstallArgo, err := canInstallArgocd(ctx, kubeClient, defaultArgoNamespace)
+			if err != nil {
+				return err
+			}
+			if canInstallArgo {
+				fmt.Println("Cannot find argocd-server deployment. Argocd may not be installed")
+				shouldInstallArgo := true
+				if !noConfirm {
+					shouldInstallArgo = cli.AskToProceed("argo-cd not found, possibly not installed. Proceed to install? [y/n]")
```
Comment (cruizen) [inline cmd/initialize/init.go:95]:
```suggestion
					shouldInstallArgo = cli.AskToProceed("ArgoCD not found, possibly not installed. Proceed to install? [y/n]")
```

**Review by cruizen — APPROVED** [review]
Looks good. Couple of minor nitpicks in comments.

File: .github/workflows/trivy.yml (hunk 3 above)
Comment (Rohitrajak1807) [reply .github/workflows/trivy.yml:6]: Yeah build sounded confusing(we had 3 by that name)

File: config/embed.go (hunk 2 above)
Comment (Rohitrajak1807) [reply config/embed.go:6]: Will add an issue

Comment (Rohitrajak1807) [reply config/embed.go:6]: https://github.com/arlonproj/arlon/issues/301

**Pull request merged**